src/panel/app_panel.js

```diff
@@ -93,4 +97,81 @@ export default class AppPanel {
       this.minify();
     }
   }
+
+  async setPoi(poi, options = {}) {
+    this.activePoiId = poi.id;
+    this.panels.forEach(panel => {
+      if (panel.isPoiCompliant) {
+        panel.setPoi(poi, options);
+      } else if (!options.isFromList && !options.isFromFavorite) {
+        panel.close();
+      }
+    });
+    this.unminify();
+  }
+
+  async loadPoiById(id, options) {
+    if (id) {
+      const poi = await ApiPoi.poiApiLoad(id);
+      if (poi) {
+        this.setPoi(poi, options);
+      } else {
+        this.resetLayout();
+      }
+      return poi;
+    } else {
+      this.resetLayout();
+    }
+  }
+
+  unsetPoi() {
+    this.activePoiId = null;
+  }
+
+  emptyClickOnMap() {
+    this.panels.forEach(p => {
+      if (p.emptyClickOnMap) {
+        p.emptyClickOnMap();
+      }
+    });
+  }
+
+  _openPanel(panelToOpen, options) {
+    /*
+      "unminify" needs to be called before panel.open :
+      DirectionPanel will minify the main search input (unused for Directions)
+    */
+    this.unminify();
+    this.panels.forEach(panel => {
+      if (panel === panelToOpen) {
+        panel.open(options);
+      } else {
+        panel.close();
+      }
+    });
+  }
+
+  openDirection(options) {
+    this._openPanel(this.directionPanel, options);
+  }
+
+  openFavorite() {
+    this._openPanel(this.favoritePanel);
+  }
+
+  openCategory(options) {
+    this._openPanel(this.categoryPanel, options);
+  }
+
+  resetLayout() {
+    this._openPanel(this.servicePanel);
+  }
+
+  async keepOnlyPoi() {
```

Comment:
this function is unused now and can be removed

Reply:
Done.